### PR TITLE
fixing a bad record ordering for EnrollmentDailyRecord that was causi…

### PR DESCRIPTION
**Pull request by azafty468**
…ng fields to be incorrectly read

### CODE REVIEW

**Review by brianhw — APPROVED** [review]
LGTM. 👍

**Comment by brianhw** [comment]
I also kicked off a manual acceptance test on Jenkins --  see edx-analytics-pipeline-acceptance-test-manual/1470.   The enrollment tests are still failing, but are closer.   Now they just differ in date format.  For example:

First has 1, Second has 0:  (u'2014-08-01', u'edX/Open_DemoX/edx_demo_course', 3, 4)
First has 0, Second has 1:  (datetime.date(2014, 8, 1), 'edX/Open_DemoX/edx_demo_course', 3, 4)

This is a side effect of switching to a Record, I suspect.  Abdul ran into a similar situation with student_engagement.

**Review by brianhw — CHANGES_REQUESTED** [review]
It would be good to fix the acceptance test first.   The problem is that these date fields are defined as StringField, but they should be defined as DateField.  The call to get_hive_schema() should return 'STRING', yes, but the call to .get_sql_schema()  should return 'DATE NOT NULL' as before, not 'VARCHAR(255) NOT NULL' which this produces.  It just happened to affect this table first (in validate_base()), but I believe all the other tables similarly defined would also have failed.

File: edx/analytics/tasks/insights/enrollments.py [hunk 1]
```diff
@@ -504,7 +504,7 @@ def change_state(self):
 
 class CourseEnrollmentRecord(Record):
     """A user's enrollment history."""
-    date = StringField(length=255, nullable=False, description='Enrollment date.')
+    date = DateField(nullable=False, description='Enrollment date.')
     course_id = StringField(length=255, nullable=False, description='The course the learner is enrolled in.')
     user_id = IntegerField(description='The user_id of the learner.')
     at_end = BooleanField(description='An indicator if the learner is still enrolled in the course by this date.')
```
Comment (brianhw) [inline edx/analytics/tasks/insights/enrollments.py]: nit:   "by this date" => "at the end of this date"

Comment (azafty468) [reply edx/analytics/tasks/insights/enrollments.py]: done and squashed back down to 1 commit

**Comment by azafty468** [comment]
merging.  Actually, I it says @brianhw is requesting changes in the status box at the bottom of this history

**Review by brianhw — APPROVED** [review]
Make this official.

**Pull request merged**